ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+# Keep in sync with ncm-xrootd
+# See http://www.perlmonks.org/?node_id=1072731 for the exporting method
+#use warnings;
+#my %rule_constants;
+#BEGIN {
+#  %rule_constants = (
+#    "LINE_FORMAT_PARAM" => 1,
+#    "LINE_FORMAT_ENVVAR" => 2,
+#    "LINE_FORMAT_XRDCFG" => 3,
+#    "LINE_FORMAT_XRDCFG_SETENV" => 4,
+#    "LINE_FORMAT_XRDCFG_SET" => 5,
+#    "LINE_VALUE_AS_IS" => 0,
+#    "LINE_VALUE_BOOLEAN" => 1,
+#    "LINE_VALUE_HOST_LIST" => 2,
+#    "LINE_VALUE_INSTANCE_PARAMS" => 3,
+#    "LINE_VALUE_ARRAY" => 4,
+#    "LINE_VALUE_HASH_KEYS" => 5,
+#    "LINE_VALUE_STRING_HASH" => 6,
+#    "LINE_VALUE_OPT_NONE" => 0,
+#    "LINE_VALUE_OPT_SINGLE" => 1,
+#    "LINE_FORMAT_DEFAULT" => LINE_FORMAT_PARAM,
+#    "LINE_QUATTOR_COMMENT" => "\t\t# Line generated by Quattor",
+#    "LINE_OPT_DEF_REMOVE_IF_UNDEF" => 0,
+#    "LINE_OPT_DEF_ALWAYS_RULES_ONLY" => 0,
+#  );
+#}
+
+#use constant \%rule_constants;
+
+use base 'Exporter';
+
+my @rule_constants = ('LINE_FORMAT_PARAM',
+                   'LINE_FORMAT_ENVVAR',
+                   'LINE_FORMAT_XRDCFG',
+                   'LINE_FORMAT_XRDCFG_SETENV',
+                   'LINE_FORMAT_XRDCFG_SET',
+                   'LINE_VALUE_AS_IS',
+                   'LINE_VALUE_BOOLEAN',
+                   'LINE_VALUE_HOST_LIST',
+                   'LINE_VALUE_INSTANCE_PARAMS',
+                   'LINE_VALUE_ARRAY',
+                   'LINE_VALUE_HASH_KEYS',
+                   'LINE_VALUE_STRING_HASH',
+                   'LINE_VALUE_OPT_NONE',
+                   'LINE_VALUE_OPT_SINGLE',
+                   'LINE_FORMAT_DEFAULT',
+                   'LINE_QUATTOR_COMMENT',
+                   'LINE_OPT_DEF_REMOVE_IF_UNDEF',
+                   'LINE_OPT_DEF_ALWAYS_RULES_ONLY',
+                  );
+our @EXPORT_OK;
+our %EXPORT_TAGS;
+push @EXPORT_OK, @rule_constants;
+$EXPORT_TAGS{rule_constants} = \@rule_constants;
+
+
+# Backup file extension
+use constant BACKUP_FILE_EXT => ".old";
+
+
+
+# This function formats an attribute value based on the value format specified.
+#
+# Arguments :
+#        attr_value : attribue value
+#        line_fmt : line format (see LINE_FORMAT_xxx constants)
+#        value_fmt : value format (see LINE_VALUE_xxx constants)
+sub formatAttributeValue () {
+  my $function_name = "formatAttributeValue";
+  my $self = shift;
+
+  my $attr_value = shift;
+  unless ( defined($attr_value) ) {
+    $self->error("$function_name: 'attr_value' argument missing");
+    return 1;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'list_fmt' argument missing");
+    return 1;
+  }
+  my $value_fmt = shift;
+  unless ( defined($value_fmt) ) {
+    $self->error("$function_name: 'value_fmt' argument missing");
+    return 1;
+  }
+
+  $self->debug(2,"$function_name: formatting attribute value >>>$attr_value<<< (line fmt=$line_fmt, value fmt=$value_fmt)");
+
+  my $formatted_value;
+  if ( $value_fmt == LINE_VALUE_HOST_LIST ) {    
```

Comment:
probably need a lookup table, like https://github.com/quattor/CCM/blob/master/src/main/perl/TextRender.pm#L34
the disadvantage is that you can't log in the methods, but they can be unittested individually, so that's not not such an issue